app/workers/account_login_statistics_job.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class AccountLoginStatisticsJob
+  include Sidekiq::Worker
+
+  def perform
+    total_stats.each do |metric, count|
+      StatsD.gauge(metric, count)
+    end
+  end
+
+  private
+
+  def total_stats
+    execute_sql(
+      account_login_stats_sql,
+      1.year.ago,
+      1.month.ago,
+      1.week.ago,
+      1.day.ago
+    ).first
+  end
+
+  def execute_sql(sql, *args)
+    ActiveRecord::Base.connection_pool.with_connection do |c|
+      c.raw_connection.exec_params(sql, args).to_a
+    end
+  end
+
+  def account_login_stats_sql
+    <<-SQL.squish
+      SELECT
+        #{count_column_sql_statements}
+      FROM account_login_stats
+    SQL
+  end
+
+  def count_column_sql_statements
+    AccountLoginStat::LOGIN_TYPES.map do |type|
+      %(
+        COUNT(#{type}_at IS NOT NULL) AS "account_login_stats.total_#{type}_accounts",
```

Comment:
I wonder if it would be clearer to do this within `ActiveRecord` instead of invoking direct SQL. Here's a thought of how it'd look:
`login_columns = login_types.map do |type| (type+'_at').to_sym end
week_count = login_columns.map do |t| AccountLoginStat.where(t => 1.week.ago..) end`
month_count = login_columns.map do |t| AccountLoginStat.where(t => 1.month.ago..) end`
`etc`
Where `login_column` is creating an array of symbols representing the columns (`:idme_at, :myhealthevet_at, :dslogon_at`)
And then the extra dot at the end of `1.month.ago..` `1.week.ago..` represents a time frame with infinity operator, technically '1 month ago to infinity in the future', but realistically, '1 month ago to now'

Reply:
If you wanna keep it raw SQL, since it's a Sidekiq job without any job parameters, I'm not super concerned or anything, so fine with keeping it as well

Reply:
Yeah, I did worry a little about the readability of this. That being said, I went this direction for a few reasons:

1) It's more performant as it only requires hitting the DB once.

2) It returns a hash result that's already formatted in a way that's fairly easy for StatsD to consume, so it saves me the trouble of having to incorporate a step to gather all the separate counts and build a hash with them. That wouldn't be terribly complicated but I almost feel like this is simpler.

Reply:
Good enough reason 4 me